src/Features/Admin/Vendors/EditVendor/index.js

```diff
@@ -77,34 +122,63 @@ const EditVendor = () => {
       },
     },
     {
-      type: TEXT_FIELD,
-      textFieldType: 'text',
-      label: 'Timing',
-      variant: 'standard',
-      value: '',
-      errorMessage: '',
-      onChange: ({ target: { value } }, index) => {
-        const updatedFields = fieldChangeHandler(fields, value, index);
-        setFields(updatedFields);
-      },
-    },
-    {
-      type: MULTI_SELECT,
+      type: SELECT,
       label: 'Building',
       values: ['Main', 'Cherry', 'Qasre Sheeren'],
       value: [],
+      variant: 'standard',
+
+      name: 'building',
       errorMessage: '',
       onChange: ({ target: { value } }, index) => {
         const updatedFields = fieldChangeHandler(fields, value, index);
         setFields(updatedFields);
       },
     },
   ]);
+  const [vendor, setVendor] = useState('');
 
+  const { data: vendorById } = FetchUserById(id);
+
+  useEffect(() => {
+    if (vendorById !== undefined) {
+      setVendor(vendorById);
+      fields.map((field) => {
+        field.value = vendorById[field.name];
+
+        // if (field.label === 'Email') {
+        //   console.log(vendorById[field.name]);
+        //   field.value = vendorById.email;
+        //   console.log('value', field.value);
+        // } else if (field.label === 'Name') {
+        //   console.log(vendorById[field.name]);
+        //   field.value = vendorById.name;
+        //   console.log('value', field.value);
+        // } else if (field.label === 'Password') {
+        //   console.log(vendorById[field.name]);
+        //   field.value = vendorById.password;
+        //   console.log('value', field.value);
+        // }
+      });
+      setFields(fields);
+    }
+  }, [vendorById]);
   const saveHandler = () => {
     const { validateArray, isValid } = validateOnSubmit(fields);
     setFields(validateArray);
-    isValid ? setOnSaveSuccess(true) : setOnSaveSuccess(false);
+
+    if (isValid) {
+      const vendorData = {};
+      fields.map(({ name, value }) => {
+        if (name !== 'building' && name !== 'contact' && name !== 'role') {
+          vendorData[name] = value;
+        }
+      });
+      EditVendor.mutateAsync({ id, token, vendorData });
+      setOnSaveSuccess(true);
```

Comment:
Mutation does have loading flag so why do we need to to set the state separately ??